### PR TITLE
[ENH] Implement pure-Python ICA with mixture modeling

### LINKED ISSUE #2: Remove non-Python dependencies
The current version uses `fslmaths` for a number of steps that could very easily be done with `nilearn` or `nibabel` + `numpy`.

The only difficult part that I can think of is handling MELODIC's mixture modeling. We can either (1) assume the MELODIC data are in BIDS format already and work from there or (2) try to "implement" the mixture modeling method with a Python-based ICA algorithm.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #2.

TODO:

- [x] Incorporate license and attribution info for `nipy`'s GGM.
- [x] Convert pseudocode to real code (this mostly means supporting images in addition to arrays and getting `computefeats2` working).
- [ ] Compare mixture modeling results to MELODIC.
- [ ] Add tests.
- [x] Integrate changes into workflow.

Changes proposed in this pull request:

- Add GGM class from `nipy` for mixture modeling.
- Use `tedana` for PCA + ICA.

### CODE REVIEW

File: aroma/utils.py [hunk 1]
```diff
@@ -5,138 +5,67 @@
...
+from tedana.decomposition import tedica, ma_pca
+from tedana.utils import get_spectrum
+from tedana.stats import computefeats2
 
+from .mixture import GGM
 
-def runICA(fsl_dir, in_file, out_dir, mel_dir_in, mask, dim, TR):
-    """Run MELODIC and merge the thresholded ICs into a single 4D nifti file.
+
+def runICA(in_file, mask, n_components, TR):
```
Comment (tsalo) [inline aroma/utils.py]: @eurunuela @CesarCaballeroGaudes I would love to get your thoughts on this approach, even though it's not currently functional.

File: aroma/aroma.py [hunk 2]
```diff
@@ -152,51 +154,35 @@ def aroma_workflow(
...
-                " - No example_func was found in the Feat directory. "
-                "A mask will be created including all voxels with varying "
-                "intensity over time in the fMRI data. Please check!\n"
-            )
-        math_command = "{0} {1} -Tstd -bin {2}".format(
-            op.join(fsl_dir, "fslmaths"), in_file, new_mask
-        )
-        os.system(math_command)
+    if not mask:
+        mask = masking.compute_epi_mask(in_file)
```
Comment (tsalo) [inline aroma/aroma.py:158]: More aggressive masking seems to be necessary. I noticed that the ICA components were showing up outside the brain with the old method.

**Comment by eurunuela** [comment]
As discussed with @CesarCaballeroGaudes yesterday, the mixture modeling is not necessary but it helps with the accuracy.

The ICA-AROMA paper states:

> To increase accuracy of the spatial features, IC spatial maps are first thresholded using a Gamma/Gaussian alternative testing approach (p N 0.5). This approach is automated in FSL's MELODIC.

So, we could try running the workflow without the mixture modeling first and see if the results look good. If we see the mixture modeling is critical for accuracy, we can open a new PR with it.

**Comment by tsalo** [comment]
To be honest, I'm not confident that either the features or the classifier in AROMA are robust enough to handle inputs that don't exactly match the original method. From a conceptual standpoint, unthresholded maps may be fine, but from a practical one, I'm not so sure.

**Comment by tsalo** [comment]
I have https://github.com/Brainhack-Donostia/aroma/pull/22 open now, so I'm going to close this.

**Pull request closed without merging**